### PR TITLE
Add CONTRIBUTING guide

### PR DESCRIPTION
**Pull request by icy-arctic-fox**
Necessary for Hacktoberfest - #17

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+- [Processing](https://processing.org/)
+  - [API Reference](https://processing.org/reference)
+- [Crystal](https://crystal-lang.org/)
+  - [Crystal API](https://crystal-lang.org/api/)
+
+## How can I contribute?
+
+This project is still in its early stages.
+Most features and the Processing API still need to be implemented.
+There's a lot to do, find something that interest you!
```
Comment (postmodern) [inline CONTRIBUTING.md]: interest**s**

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+If there isn't an issue for something you want to add or change, please create one first.
+Additionally, there is a feature checklist in the [README](README.md).
+
+### Help Wanted
+
+These are the area that maintainers are looking to tackle next.
+These are quite substantial, but assistance would be greatly appreciated!
+
+- [Mock/stub rendering](https://github.com/icy-arctic-fox/processing.cr/issues/2) - Need a way to reliably test visual aspects.
+- Implement various parts of the Processing API.
```
Comment (postmodern) [inline CONTRIBUTING.md:53]: Maybe mention that contributors can `git grep -C 1 not_implemented` to see all methods that need implementing?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+If you can't find anything already out there, submit an issue.
+Explain what you're trying to do and maybe an explanation of why.
+
+## Development
+
+To get started:
+
+1. [Fork the repository](https://github.com/icy-arctic-fox/processing/fork)
+2. Clone the forked git repository.
+3. Run `shards` in the repository to pull developer dependencies.
```
Comment (postmodern) [inline CONTRIBUTING.md]: Probably should mention how to install SDL2. I'm betting that will be a major stumbling block for new users. I can probably add `apt-get`, `dnf`, `brew` commands to the README.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+Here are some useful links:
+
+- [README](README.md)
+- [Issue tracker](https://github.com/icy-arctic-fox/processing.cr/issues)
+- [Processing](https://processing.org/)
+  - [API Reference](https://processing.org/reference)
+- [Crystal](https://crystal-lang.org/)
+  - [Crystal API](https://crystal-lang.org/api/)
+
```
Comment (postmodern) [inline CONTRIBUTING.md:33]: We should also link to [sdl.cr](/ysbaddaden/sdl.cr). I found myself constantly reviewing it's API.

Also linking to [Processing v4's source code](https://github.com/processing/processing4/tree/master/core/src/processing/core) as the reference implementation might help people understand how certain methods were implemented.

**Review by postmodern — COMMENTED** [review]
Only found on spelling mistake. Left some additional suggestions for things to add/document.

**Comment by icy-arctic-fox** [comment]
Updated with your suggestions.

File: CONTRIBUTING.md (hunk 4 above)
Comment (icy-arctic-fox) [reply CONTRIBUTING.md:33]: Totally agree, I reference those often.

File: CONTRIBUTING.md (hunk 3 above)
Comment (icy-arctic-fox) [reply CONTRIBUTING.md]: SDL's website has a decent guide but can be overwhelming to newbies.

File: CONTRIBUTING.md (hunk 2 above)
Comment (icy-arctic-fox) [reply CONTRIBUTING.md:53]: That's a handy command!

**Pull request merged**